servers/quarkus-auth/src/main/java/org/projectnessie/server/authz/CelBatchAccessChecker.java

```diff
@@ -77,65 +81,48 @@ private void canPerformOp(Check check, Map<Check, String> failed) {
-    ImmutableMap<String, Object> arguments =
-        ImmutableMap.of(
-            "ref",
-            refName,
-            "role",
-            roleName,
-            "op",
-            check.type().name(),
-            "path",
-            "",
-            "contentType",
-            "");
+    String role = getRoleName();
+    String op = check.type().name();
+    String ref = Optional.ofNullable(check.ref()).map(NamedRef::getName).orElse("");
+    Map<String, Object> arguments =
+        ImmutableMap.of("ref", ref, "role", role, "op", op, "path", "", "contentType", "");
 
     Supplier<String> errorMsgSupplier =
         () ->
             String.format(
-                "'%s' is not allowed for role '%s' on reference '%s'",
-                check.type(), roleName, refName);
+                "'%s' is not allowed for role '%s' on reference '%s'", check.type(), role, ref);
     canPerformOp(arguments, check, errorMsgSupplier, failed);
   }
 
   private void canPerformOpOnPath(Check check, Map<Check, String> failed) {
-    String roleName = getRoleName();
-    Content.Type contentType = check.contentType();
-    String contentKeyPathString = check.key().toPathString();
-    ImmutableMap<String, Object> arguments =
+    String role = getRoleName();
+    String op = check.type().name();
+    String contentType = Optional.ofNullable(check.contentType()).map(Type::name).orElse("");
+    String path = Optional.ofNullable(check.key()).map(ContentKey::toPathString).orElse("");
+    String ref = Optional.ofNullable(check.ref()).map(NamedRef::getName).orElse("");
+    Map<String, Object> arguments =
         ImmutableMap.of(
-            "ref",
-            check.ref().getName(),
-            "path",
-            contentKeyPathString,
-            "role",
-            roleName,
-            "op",
-            check.type().name(),
-            "contentType",
-            contentType != null ? contentType.name() : "");
+            "ref", ref, "path", path, "role", role, "op", op, "contentType", contentType);
 
     Supplier<String> errorMsgSupplier =
-        () ->
-            String.format(
-                "'%s' is not allowed for role '%s' on content '%s'",
-                check.type(), roleName, contentKeyPathString);
+        () -> String.format("'%s' is not allowed for role '%s' on content '%s'", op, role, path);
 
     canPerformOp(arguments, check, errorMsgSupplier, failed);
   }
 
   private void canPerformRepositoryConfig(Check check, Map<Check, String> failed) {
-    RepositoryConfig.Type repositoryConfigType = requireNonNull(check.repositoryConfigType());
+    String op = check.type().name();
+    String type =
+        Optional.ofNullable(check.repositoryConfigType())
+            .map(RepositoryConfig.Type::name)
+            .orElse("");
 
-    ImmutableMap<String, Object> arguments = ImmutableMap.of("type", repositoryConfigType.name());
+    Map<String, Object> arguments = ImmutableMap.of("type", type, "op", op);
```

Comment:
That came as a surprise to me BTW: the `op` argument was missing, which makes the default rule for viewing refs fail. I think it's a bug.